examples/plot_localizer_analysis_permutation.py

```diff
@@ -0,0 +1,148 @@
+   then contrast estimation)
+5. Permutation tests are performed to obtain uncorrected and corrected p values
+at each voxel for a contrast comparing only two conditions.
+
+"""
+
+print(__doc__)
+
+from os import mkdir, path
+
+import numpy as np
+import pandas as pd
+from nilearn import plotting
+
+from nistats.glm import FirstLevelGLM
+from nistats.design_matrix import make_design_matrix
+from nistats import datasets
+from nistats.permutation_tests import first_level_permutation_test
+from nistats.utils import z_score
+import nibabel as nib
+
+
+n_proc = 6
+
+### Data and analysis parameters #######################################
+
+# timing
+n_scans = 128
+tr = 2.4
+frame_times = np.linspace(0.5 * tr, (n_scans - .5) * tr, n_scans)
+
+# data
+data = datasets.fetch_localizer_first_level()
+paradigm_file = data.paradigm
+fmri_img = data.epi_img
+
+### Design matrix ########################################
+
+paradigm = pd.read_csv(paradigm_file, sep=' ', header=None, index_col=None)
+paradigm.columns = ['session', 'name', 'onset']
+design_matrix = make_design_matrix(
+    frame_times, paradigm, hrf_model='canonical with derivative',
+    drift_model="cosine", period_cut=128)
+
+### Perform a GLM analysis ########################################
+
+fmri_glm = FirstLevelGLM().fit(fmri_img, design_matrix)
+
+### Estimate contrasts #########################################
+
+# Specify the contrasts
+contrast_matrix = np.eye(design_matrix.shape[1])
+contrasts = dict([(column, contrast_matrix[i])
+                  for i, column in enumerate(design_matrix.columns)])
+
+contrasts["audio"] = contrasts["clicDaudio"] + contrasts["clicGaudio"] +\
+    contrasts["calculaudio"] + contrasts["phraseaudio"]
+contrasts["video"] = contrasts["clicDvideo"] + contrasts["clicGvideo"] + \
+    contrasts["calculvideo"] + contrasts["phrasevideo"]
+contrasts["computation"] = contrasts["calculaudio"] + contrasts["calculvideo"]
+contrasts["sentences"] = contrasts["phraseaudio"] + contrasts["phrasevideo"]
+
+# Short list or more relevant contrasts
+# contrasts = {
+#     "left-right": (contrasts["clicGaudio"] + contrasts["clicGvideo"]
+#                    - contrasts["clicDaudio"] - contrasts["clicDvideo"]),
+#     "H-V": contrasts["damier_H"] - contrasts["damier_V"],
+#     "audio-video": contrasts["audio"] - contrasts["video"],
+#     "video-audio": -contrasts["audio"] + contrasts["video"],
+#     "computation-sentences": (contrasts["computation"] -
+#                               contrasts["sentences"]),
+#     "reading-visual": contrasts["phrasevideo"] - contrasts["damier_H"]
+#     }
+
+contrasts = {
+    "reading-visual": contrasts["phrasevideo"] - contrasts["damier_H"]
+    }
+
+### Permutation tests ###########################################
+
+# Design matrix obj arrangement for paradigm permutations
+# What about creating design_matrix class to wrap make design matrix and
+# respective parameters. Then could just pass that object
+
```

Comment:
What do you mean by not needing an object? that I could instead say contrasts['reading-visual'] = ....